addons/website/static/src/builder/plugins/customize_website_plugin.js

```diff
@@ -0,0 +1,698 @@
+            this.pendingThemeRequests = this.pendingThemeRequests.filter(
+                (req) => req.isViewData !== isViewData || req.shouldReset !== shouldReset
+            );
+            if (!aggregatedToEnable.size && !aggregatedToDisable.size) {
+                return;
+            } else {
+                rpc("/website/theme_customize_data", {
+                    is_view_data: isViewData,
+                    enable: [...aggregatedToEnable],
+                    disable: [...aggregatedToDisable],
+                    reset_view_arch: shouldReset,
+                })
+                    .then(() => Promise.all(defs.map((def) => def.resolve())))
+                    .catch(() => Promise.all(defs.map((def) => def.reject())));
+            }
+        }, 0);
+        return def;
+    }
+
+    getConfigKey(key) {
+        if (key.startsWith("!")) {
+            return !this.activeRecords[key.substring(1)];
+        }
+        return this.activeRecords[key];
+    }
+    withCustomHistory(action) {
+        const applyFn = action.apply;
+        const apply = async (arg) => {
+            const oldValue = action.getValue(arg);
+            const { value } = arg;
+            const blockedApply = (v) => {
+                this.services.ui.block({ delay: 2500 });
+                return applyFn({ ...arg, value: v })
+                    .then(() => {
+                        this.dispatchTo("trigger_dom_updated");
+                    })
+                    .finally(() => this.services.ui.unblock());
+            };
+            await blockedApply(value);
+            this.dependencies.history.addCustomMutation({
+                apply: () => blockedApply(value),
+                revert: () => blockedApply(oldValue),
+            });
+        };
+        return { preview: false, ...action, apply };
+    }
+
+    async loadTemplateKey(key) {
+        if (!this.getTemplateKey(key)) {
+            // TODO: make a python method that can return several templates at
+            // once and batch the ORM call.
+            this.activeTemplateViews[key] = await this.services.orm.call(
+                "ir.ui.view",
+                "render_public_asset",
+                [`${key}`, {}]
+            );
+        }
+        return this.getTemplateKey(key);
+    }
+    toggleTemplate(action, apply) {
+        if (!apply) {
+            // Empty the container and restore the original content
+            action.editingElement.replaceChildren(this.beforePreviewNodes);
+            this.beforePreviewNodes = null;
+            return;
+        }
+
+        if (!this.beforePreviewNodes) {
+            // We are about to apply a template on non-previewed content,
+            // save that content's nodes.
+            this.beforePreviewNodes = [...action.editingElement.childNodes];
+        }
+
+        // Empty the container and add the template content
+        const templateFragment = parseHTML(this.document, this.getTemplateKey(action.params.view));
```

Comment:
This can be done in a follow up PR, but we should update `parseHTML` to be markup aware in terms of its `html` param (for example, by using `createDocumentFragmentFromContent` internally) so that we can remove the ci flag from it.